index.js

```diff
@@ -1,6 +1,10 @@
 module.exports = filterObjectOnSchema;
 
 
+function isObject(obj) {
+    return obj === Object(obj);
```

Comment:
technically true, but potentially churns an object, obj && typeof obj === 'object' is equivalent.